src/Bicep.LangServer/Utils/DocumentSelectorFactory.cs

```diff
@@ -1,27 +1,45 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT License.
+
 using Bicep.Core;
+using Bicep.LanguageServer.Settings;
 using OmniSharp.Extensions.LanguageServer.Protocol.Models;
+using Azure.Deployments.Core.Extensions;
+using Bicep.LanguageServer.Options;
 
 namespace Bicep.LanguageServer.Utils
 {
-    public class DocumentSelectorFactory
+    public class DocumentSelectorFactory(IBicepLangServerOptions? langServerOptions)
```

Comment:
To centralize the defaults for lang server options, could we just make this non-null and force the caller to pass in an instance? See my other comment on using a record - I think that simplifies the boilerplate, as you can just write `new DocumentSelectorFactory(new())` or `new DocumentSelectorFactory(new() with { VsCompatibilityMode = true });` in test setup code (rather than `new DocumentSelectorFactory(null)`)